### PR TITLE
CNS-805-conflict-module-readme

### PR DESCRIPTION
**Pull request by Yaroms**

### CODE REVIEW

File: x/conflict/README.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+### Finalization Conflict
+N/A
+
+
+### Self Provider Conflict
+N/A
+
+
+### Response Conflict
+A response conflict occurs when a consumer receives mismatched responses from different providers. In such cases, the consumer is eligible to send a conflict detection message, which includes the relay request and responses from the two providers. This conflict detection message is then validated to ensure that the responses are different, the signatures match the providers and consumer, and that the API is deterministic. If the message is valid, a conflict is opened.
```
Comment (oren-lava) [inline x/conflict/README.md:34]: you didn't explain how come a consumer asks two providers the same query (normally, the consumer talks to one provider). If I'm not mistaken, every 16th query is checked with two providers right?

File: x/conflict/README.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+A group of validators is selected as a jury to determine the fraudulent and honest providers. Through an event, the chain announces the conflict voting period and the participating providers. During the voting period, providers need to submit their hashed response + salt to the original relay request. This is done to prevent other providers from cheating or copying their vote. Once the voting period ends, the conflict moves to the reveal state. In this state, providers need to reveal their response + salt, which is then verified and compared to the original responses. After the reveal period ends, the votes are counted, and the provider with the fewest votes is penalized by having a fraction of their staked tokens taken and distributed among all the participants.
```
Comment (oren-lava) [inline x/conflict/README.md:36]: regarding the penalty for losing the conflict, don't also the ones that voted for the losing provider get penalized too?

File: x/conflict/README.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+### Commit Period
+
+This is the voting period of the conflict, providers that are selected as jury are required to submit their hashed vote.
+The hash consisnt of nonce + response hash + provider address
```
Comment (oren-lava) [inline x/conflict/README.md:41]: end with "."

File: x/conflict/README.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+type MsgConflictVoteCommit struct {
+	Creator string // provider address
+	VoteID  string // the conflict unique id
+	Hash    []byte // the hashed nonce + response + provider address
+}
+```
+
+### Reveal Period
+
+Once the commit period ends, the conflict enters reveal state. Here providers need to submit their full response and nonce. this is than validated and compared to the original providers responses. The result of the vote (Provider A, Provider B, None) is than saved on state, waiting for the end of the conflict.
```
Comment (oren-lava) [inline x/conflict/README.md:53]:
```suggestion
Once the commit period ends, the conflict enters the reveal state. At this point, providers are required to submit their full response and nonce. These responses are then validated and compared to the original providers' responses. The result of the vote (Provider A, Provider B, None) is then saved in the state, awaiting the end of the conflict.
```

File: x/conflict/README.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+	Creator string // provider address
+	VoteID  string // the conflict unique id
+	Nonce   int64  // random nonce
+	Hash    []byte // hash of the response
+}
+```
+
+### Conflict Resolve
+
+Once the reveal period ends votes are than counted to determine the results of the conflict. Votes are weighted by providers stake.
```
Comment (oren-lava) [inline x/conflict/README.md:66]:
```suggestion
Once the reveal period ends, votes are counted to determine the results of the conflict. The votes are weighted based on the providers' stake. Providers on the jury who did not vote will be punished with jail and slashing, and the slashed amount will be added to the conflict reward pool.

To resolve the conflict, a majority of votes must be in favor of Provider A, Provider B, or None of them. If a majority is not reached, the conflict reward pool is given to the consumer who reported the conflict.

Once a majority is reached, providers who voted for the wrong side of the conflict will be slashed and frozen, and the slashed amount will be added to the conflict reward pool. The reward pool is then distributed between the consumer and the providers who voted for the correct provider.
```

File: x/conflict/README.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+The conflict module contains the following parameters:
+
+| Key                                    | Type                    | Default Value    |
+| -------------------------------------- | ----------------------- | -----------------|
+| MajorityPercent                        | math.LegacyDec          | 0.95              |
+| VoteStartSpan                              | uint64          | 3             |
+| VotePeriod                       | uint64          | 2                |
+| Rewards                        | rewards                  | N/A                |
+
+`MajorityPercent` majority needed to conclude who is the winner of the conclict
```
Comment (oren-lava) [inline x/conflict/README.md:84]: -> `MajorityPercent` determines the majority needed to conclude who is the winner of the conflict.

File: x/conflict/README.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+`VoteStartSpan` is the amount of epochs in the past that a consumer can send conflict detection message for
```
Comment (oren-lava) [inline x/conflict/README.md:85]: -> `VoteStartSpan` is the number of epochs in the past that a consumer can send a conflict detection message for.

File: x/conflict/README.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+`VotePeriod` is the amount of epochs for each voting state (commit/reveal)
```
Comment (oren-lava) [inline x/conflict/README.md:86]: -> `VotePeriod` is the number of epochs for each voting state (commit/reveal).

File: x/conflict/README.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+### Conflict Resolve
+
+Once the reveal period ends votes are than counted to determine the results of the conflict. Votes are weighted by providers stake. 
+Providers in the jury that did not vote are punished by jail and slashing, their slashed amount will be added to the conflict reward pool.
+For the conflict resolution there needs to be a majority met of votes for Provider A, Provider B or None of them. 
+If a majority was not met, conflict reward pool is given to the consumer that reported the conflict.
+Once a majority is met providers that voted to the wrong side of the conflict are slashed and frozen, the slashed amount is added to the conflict reward pool.
+Now the reward pool is distributed between the comsumer and the providers that voted for the correct provider.
+
+## Parameters
```
Comment (oren-lava) [inline x/conflict/README.md:73]: when explaining about the parameters below, put another new line between them so in the README each will be in a new line

File: x/conflict/README.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+| -------------------------------------- | ----------------------- | -----------------|
+| MajorityPercent                        | math.LegacyDec          | 0.95              |
+| VoteStartSpan                              | uint64          | 3             |
+| VotePeriod                       | uint64          | 2                |
+| Rewards                        | rewards                  | N/A                |
+
+`MajorityPercent` majority needed to conclude who is the winner of the conclict
+`VoteStartSpan` is the amount of epochs in the past that a consumer can send conflict detection message for
+`VotePeriod` is the amount of epochs for each voting state (commit/reveal)
+`Rewards` defines how the reward pool tokens will be distributed between all participants
```
Comment (oren-lava) [inline x/conflict/README.md:87]: -> ... between all the participants:

File: x/conflict/README.md [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+
+```go
+type Rewards struct {
+	WinnerRewardPercent github_com_cosmos_cosmos_sdk_types.Dec // the winnig provider of the conflict portion
```
Comment (oren-lava) [inline x/conflict/README.md:91]: -> the conflict's winner portion (provider)

File: x/conflict/README.md [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+	ClientRewardPercent github_com_cosmos_cosmos_sdk_types.Dec // the reporter of the conflict portion
```
Comment (oren-lava) [inline x/conflict/README.md:92]: -> the conflict's reporter portion

File: x/conflict/README.md [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+}
+```
+## Queries
+
+The Conflict module supports the following queries:
+
+| Query             | Arguments         | What it does                                  |
+| ----------        | ---------------   | ----------------------------------------------|
+| `params`          | none              | show the params of the module                 |
+| `consumer-conflicts` | consumer (string)              | shows all conflicts reported and active by the consumer         |
```
Comment (oren-lava) [inline x/conflict/README.md:103]: -> shows all the reported and active conflicts of a consumer

File: x/conflict/README.md [hunk 14]
```diff
@@ -0,0 +1,124 @@
...
+| `list-conflict-vote` | none           | shows all active conflicts                |
+| `show-conflict-vote`       | voteID (string)           | shows a specific active conflict                             |
+
+## Transactions
+
+The Conflict module transactions are not ment to be sent by the CLI and are used by the consumer and providers.
```
Comment (oren-lava) [inline x/conflict/README.md:109]: ment -> meant
to be sent by the CLI -> to be used with the CLI
consumer -> consumers

File: x/conflict/README.md [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+for more information look [here](../../proto/lavanet/lava/conflict/tx.proto)
```
Comment (oren-lava) [inline x/conflict/README.md:110]: end with "."

File: x/conflict/README.md [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+
+### Events
+
+The plans module has the following events:
```
Comment (oren-lava) [inline x/conflict/README.md:114]: plans -> conflict

File: x/conflict/README.md [hunk 17]
```diff
@@ -0,0 +1,124 @@
...
+| Event             | When it happens       |
+| ----------        | --------------- |
+| `response_conflict_detection`        | a new conflict has opened (this event includes all of the jury providers) and is entering the commit stage  |
```
Comment (oren-lava) [inline x/conflict/README.md:117]: -> A new conflict has been opened, which involves all of the jury providers. It is now entering the commit stage

File: x/conflict/README.md [hunk 18]
```diff
@@ -0,0 +1,124 @@
...
+| `conflict_vote_reveal_started`        | conflict has transitioned to reveal state  |
+| `conflict_vote_got_commit`        | provider commited his vote  |
+| `conflict_vote_got_reveal`        | provider revealed his vote  |
+| `conflict_unstake_fraud_voter`        | provider was unstaked due to conflict  |
+| `conflict_detection_vote_resolved`        | conflict was succesfully resolved  |
+| `conflict_detection_vote_unresolved`        | conflict was not resolved (did not reach majority)  |
+| `spec_add`        | a successful addition of a spec  |
```
Comment (oren-lava) [inline x/conflict/README.md:124]: is this supposed to be here?

**Review by oren-lava — DISMISSED** [review]

File: x/conflict/README.md (hunk 18 above)
Comment (Yaroms) [reply x/conflict/README.md:128]: nope

File: x/conflict/README.md (hunk 6 above)
Comment (Yaroms) [reply x/conflict/README.md:88]: done

File: x/conflict/README.md (hunk 7 above)
Comment (Yaroms) [reply x/conflict/README.md:89]: done

File: x/conflict/README.md (hunk 1 above)
Comment (Yaroms) [reply x/conflict/README.md:34]: done

File: x/conflict/README.md (hunk 11 above)
Comment (Yaroms) [reply x/conflict/README.md:95]: done

File: x/conflict/README.md (hunk 12 above)
Comment (Yaroms) [reply x/conflict/README.md:96]: done

File: x/conflict/README.md (hunk 13 above)
Comment (Yaroms) [reply x/conflict/README.md:107]: done

File: x/conflict/README.md (hunk 14 above)
Comment (Yaroms) [reply x/conflict/README.md:113]: done

File: x/conflict/README.md (hunk 15 above)
Comment (Yaroms) [reply x/conflict/README.md:114]: done

File: x/conflict/README.md (hunk 17 above)
Comment (Yaroms) [reply x/conflict/README.md:121]: done

File: x/conflict/README.md (hunk 2 above)
Comment (Yaroms) [reply x/conflict/README.md:36]: yes they are

File: x/conflict/README.md (hunk 10 above)
Comment (Yaroms) [reply x/conflict/README.md:91]: done

**Review by Netanel94 — APPROVED** [review]

File: x/conflict/README.md [hunk 19]
```diff
@@ -9,9 +9,9 @@ The purpose of the conflict module is to ensure data reliability from providers
 
 ## Contents
 * [Concepts](#concepts)
-    * [Finalization Conflict](#finalization-conflict)
-    * [Self Provider Conflict](#self-provider-conflict)
     * [Response Conflict](#response-conflict)
+	* [Finalization Conflict](#finalization-conflict)
```
Comment (oren-lava) [inline x/conflict/README.md:13]: align it to the other items in the list.
If it's supposed to be sub chapter of Response conflict, add '#' to the Finalization Conflict title

**Review by shleikes — APPROVED** [review]

**Review by oren-lava — CHANGES_REQUESTED** [review]

**Pull request merged**